runners/core-construction-java/src/main/java/org/apache/beam/runners/core/construction/CoderTranslation.java

```diff
@@ -60,6 +62,9 @@ private static class DefaultTranslationContext implements TranslationContext {}
   // TODO: standardize such things
   public static final String JAVA_SERIALIZED_CODER_URN = "beam:coders:javasdk:0.1";
 
+  public static final List<String> KNOWN_EXTERNAL_CODERS =
```

Comment:
Why do we limit ourselves to a set of known coders, rather than just handle all unknown coders uniformly?

Reply:
I think we want to differentiate the primary use-case this PR is handling (consecutive external transforms) from actual unknown coders used in the Java side. For Java unknown coders we should just keep the existing failure path. Also, I think the set of coders we need to cover here is a small list (one/few per SDK) since we are primarily trying to make sure that the pipeline doesn't fail to expand when external SDKs end up using the default coder.

Reply:
I'm still not following this. What "actual unknown coders used in the Java side" would we have that we want to preserve the existing failure paths for. (And why?) If we're trying to make sure the pipeline doesn't fail when using external SDKs it doesn't seem like the right general approach to hardcode a subset of coders a that particular external SDK currently uses, even if this list is slowly changing.

Reply:
I meant any coder proto that would have hit "Unknown Coder URN %s. Known URNs: %s" here. I did not want to override the current behavior (though I'm not sure if we actually hit this in practice).

Apart from that, if we want to override the behavior for all known coders I can go ahead and do that.

Reply:
Yes, I think it's fine to override the behavior for all unknown coders.